### ISSUE TITLE
bug(typography): body-medium and body-small line heights are not accessible

### ISSUE THREAD

**Issue by kainiedziela** [angular/components#31176]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

To meet criterion 1.4.12 of WCAG 2.1 https://www.w3.org/WAI/WCAG21/Understanding/text-spacing.html body-medium and body-small type should have higher line heights

Current body-medium line height: 20px
Accessible body-medium line height for the same font size: 22px

Current body-small line height: 16px
Accessible body-small line height for the same font size: 18px

### Expected Behavior

Accessible line heights for default typography

### Actual Behavior

Inaccessible line heights for body-medium and body-small typography

### Environment

- Angular: 18
- Material: 3
- Browser(s): Any
- Operating System (e.g. Windows, macOS, Ubuntu): Any
